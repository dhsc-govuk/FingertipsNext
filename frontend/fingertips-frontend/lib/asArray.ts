export const asArray = (value?: string | string[]): string[] => {
  if (value) {
    return Array.isArray(value) ? value : [value];
  }
  return [];
};
