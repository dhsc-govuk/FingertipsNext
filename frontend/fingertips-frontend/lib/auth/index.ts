import { AuthConfigFactory } from '@/lib/auth/config';
import NextAuth from 'next-auth';

export const { handlers, signIn, signOut, auth } = NextAuth(
  AuthConfigFactory.getConfig()
);
