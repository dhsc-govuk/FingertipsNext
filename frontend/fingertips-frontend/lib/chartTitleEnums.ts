export enum ChartTitlesEnum {
  LineChart = 'Indicator trends over time',
  BarChartEmbeddedTable = 'Compare areas for one time period',
  Heatmap = 'Overview of indicators and areas',
  SpineChart = 'Area profile by indicators',
  InequalitiesBarChart = 'Inequalities comparison for one time period',
  InequalitiesLineChart = 'Inequalities trends over time',
  ThematicMap = 'Compare an indicator by areas',
  PopulationPyramid = 'Population pyramid',
}
