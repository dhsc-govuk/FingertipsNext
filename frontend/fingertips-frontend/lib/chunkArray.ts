export const maxNumAreasThatCanBeRequestedAPI = 300;

export function chunkArray(
  arrayToChunk: string[],
  chunkSize: number = maxNumAreasThatCanBeRequestedAPI
) {
  const chunkedArray = [];
  for (let i = 0; i < arrayToChunk.length; i += chunkSize) {
    chunkedArray.push(arrayToChunk.slice(i, i + chunkSize));
  }
  return chunkedArray;
}
