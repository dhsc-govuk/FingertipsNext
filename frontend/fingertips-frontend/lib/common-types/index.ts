export enum Trend {
  INCREASING = 'Increasing',
  DECREASING = 'Decreasing',
  NO_SIGNIFICANT_CHANGE = 'No significant change',
  NOT_AVAILABLE = 'No recent trend data available',
}

export enum TrendCondition {
  GETTING_BETTER = 'getting better',
  GETTING_WORSE = 'getting worse',
}

export enum Direction {
  UP = 'up',
  DOWN = 'down',
  RIGHT = 'right',
  LEFT = 'left',
}
