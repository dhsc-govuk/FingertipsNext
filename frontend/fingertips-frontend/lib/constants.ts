export const siteTitle = 'Find public health data';

export const siteDescription = `${siteTitle} is a rich source of indicators across a range of health and wellbeing themes designed to support JSNA and commissioning to improve health and wellbeing, and reduce inequalities.`;

export const spaceSeparatedPattern = /^\d+(\s+\d+)+$/;
