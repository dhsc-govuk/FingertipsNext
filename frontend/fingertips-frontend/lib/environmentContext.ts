import { getEnvironmentVariable } from './utils';

export const DHSC_AI_SEARCH_USE_MOCK_SERVICE: string =
  'DHSC_AI_SEARCH_USE_MOCK_SERVICE';
export const DHSC_AI_SEARCH_SERVICE_URL: string = 'DHSC_AI_SEARCH_SERVICE_URL';
export const DHSC_AI_SEARCH_INDEX_NAME: string = 'DHSC_AI_SEARCH_INDEX_NAME';
export const DHSC_AI_SEARCH_API_KEY: string = 'DHSC_AI_SEARCH_API_KEY';
export const DHSC_AI_SEARCH_SCORING_PROFILE: string =
  'DHSC_AI_SEARCH_SCORING_PROFILE';

const loadEnvironmentContext = (): Map<string, string> => {
  console.log('Loading environment variables ...');
  const map: Map<string, string> = new Map();

  // Set required environment variables
  const mockService = getEnvironmentVariable(
    DHSC_AI_SEARCH_USE_MOCK_SERVICE
  ) as string;
  map.set(DHSC_AI_SEARCH_USE_MOCK_SERVICE, mockService);
  const useMock: boolean = mockService === 'true';

  // Following attributes are only mandatory if the mock search service is not being used.
  map.set(
    DHSC_AI_SEARCH_SERVICE_URL,
    getEnvironmentVariable(DHSC_AI_SEARCH_SERVICE_URL, !useMock) as string
  );
  map.set(
    DHSC_AI_SEARCH_INDEX_NAME,
    getEnvironmentVariable(DHSC_AI_SEARCH_INDEX_NAME, !useMock) as string
  );
  map.set(
    DHSC_AI_SEARCH_API_KEY,
    getEnvironmentVariable(DHSC_AI_SEARCH_API_KEY, !useMock) as string
  );

  // Set optional environment variables
  map.set(
    DHSC_AI_SEARCH_SCORING_PROFILE,
    getEnvironmentVariable(DHSC_AI_SEARCH_SCORING_PROFILE, false) as string
  );

  return map;
};

export const EnvironmentContext = (function () {
  let environmentMap: Map<string, string> | undefined;

  return {
    getEnvironmentMap: function () {
      if (!environmentMap) {
        environmentMap = loadEnvironmentContext();
      }
      return environmentMap;
    },
    reset: function () {
      environmentMap = undefined;
    },
  };
})();
