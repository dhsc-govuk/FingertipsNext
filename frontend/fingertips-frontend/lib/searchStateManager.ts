import { asArray } from './pageHelpers';

export enum SearchParams {
  SearchedIndicator = 'si',
  IndicatorsSelected = 'is',
  AreasSelected = 'as',
  AreaTypeSelected = 'ats',
  GroupTypeSelected = 'gts',
  ConfidenceIntervalSelected = 'cis',
}

export type SearchParamKeys = `${SearchParams}`;

const multiValueParams = [
  SearchParams.IndicatorsSelected as string,
  SearchParams.AreasSelected as string,
  SearchParams.ConfidenceIntervalSelected as string,
];

export type SearchStateParams = {
  [SearchParams.SearchedIndicator]?: string;
  [SearchParams.IndicatorsSelected]?: string[];
  [SearchParams.AreasSelected]?: string[];
  [SearchParams.AreaTypeSelected]?: string;
  [SearchParams.GroupTypeSelected]?: string;
  [SearchParams.ConfidenceIntervalSelected]?: string[];
};

const isMultiValueTypeParam = (searchParamKey: SearchParamKeys) =>
  multiValueParams.includes(searchParamKey);

export class SearchStateManager {
  private searchState: SearchStateParams;
  private searchStateParams: URLSearchParams;

  constructor(searchState: SearchStateParams = {}) {
    this.searchState = {};
    this.searchStateParams = new URLSearchParams();

    this.searchState = searchState;
  }

  private constructPath(path: string) {
    if (this.searchStateParams.size === 0) {
      return path;
    }
    return `${path}?${this.searchStateParams.toString()}`;
  }

  private addParamToPath(searchParamKey: SearchParamKeys) {
    const paramsToAdd = asArray(this.searchState[searchParamKey]);

    if (paramsToAdd.length > 0) {
      paramsToAdd.forEach((paramToAdd) => {
        this.searchStateParams.append(searchParamKey, paramToAdd);
      });
    }
  }

  public addParamValueToState(
    searchParamKey: SearchParamKeys,
    paramToAdd: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = this.searchState[searchParamKey];

      const currentParamStateAsArray = asArray(currentParamState);
      const doesParamExist = currentParamStateAsArray.find(
        (currentParam) => currentParam === paramToAdd
      );

      if (!doesParamExist) {
        const newState: SearchStateParams = {
          ...this.searchState,
          [searchParamKey]: [...currentParamStateAsArray, paramToAdd],
        };

        this.searchState = newState;
      }
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: paramToAdd,
      };

      this.searchState = newState;
    }
  }

  public removeParamValueFromState(
    searchParamKey: SearchParamKeys,
    paramToRemove?: string
  ) {
    if (isMultiValueTypeParam(searchParamKey)) {
      const currentParamState = asArray(this.searchState[searchParamKey]);

      const newParamState = currentParamState?.filter((paramValue) => {
        return paramValue !== paramToRemove;
      });

      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: newParamState,
      };

      this.searchState = newState;
    } else {
      const newState: SearchStateParams = {
        ...this.searchState,
        [searchParamKey]: undefined,
      };

      this.searchState = newState;
    }
  }

  public removeAllParamFromState(searchParamKey: SearchParamKeys) {
    this.searchState[searchParamKey] = undefined;
  }

  public static setStateFromParams(params: URLSearchParams) {
    const newState = Object.values(SearchParams).reduce<SearchStateParams>(
      (state, searchParamKey) => {
        if (isMultiValueTypeParam(searchParamKey)) {
          const paramValues = params.getAll(searchParamKey);

          if (paramValues.length > 0) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValues,
            };

            return newState;
          }
        } else {
          const paramValue = params.get(searchParamKey);

          if (paramValue) {
            const newState: SearchStateParams = {
              ...state,
              [searchParamKey]: paramValue,
            };

            return newState;
          }
        }
        return state;
      },
      {}
    );

    const searchStateManager = new SearchStateManager(newState);
    return searchStateManager;
  }

  public static setStateFromSearchStateParams(params?: SearchStateParams) {
    if (params) {
      const newState = Object.values(SearchParams).reduce<SearchStateParams>(
        (state, searchParamKey) => {
          if (isMultiValueTypeParam(searchParamKey)) {
            const paramValues = asArray(params[searchParamKey]);

            if (paramValues.length > 0) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValues,
              };

              return newState;
            }
          } else {
            const paramValue = params[searchParamKey];

            if (paramValue) {
              const newState: SearchStateParams = {
                ...state,
                [searchParamKey]: paramValue,
              };

              return newState;
            }
          }
          return state;
        },
        {}
      );

      const searchStateManager = new SearchStateManager(newState);
      return searchStateManager;
    }
    return new SearchStateManager();
  }

  public getSearchState() {
    return this.searchState;
  }

  public generatePath(path: string) {
    this.searchStateParams = new URLSearchParams();

    Object.values(SearchParams).forEach((searchParamKey) => {
      this.addParamToPath(searchParamKey);
    });

    return this.constructPath(path);
  }
}
