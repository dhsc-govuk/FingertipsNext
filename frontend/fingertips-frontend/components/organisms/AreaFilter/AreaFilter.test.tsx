import { render, screen, within } from '@testing-library/react';
import { AreaFilter } from '.';
import userEvent from '@testing-library/user-event';
import { SearchParams } from '@/lib/searchStateManager';
import {
  mockAreaDataForNHSRegion,
  mockAreaTypes,
  mockAvailableAreas,
} from '@/mock/data/areaData';

const mockPath = 'some-mock-path';
const mockReplace = jest.fn();

jest.mock('next/navigation', () => {
  const originalModule = jest.requireActual('next/navigation');

  return {
    ...originalModule,
    usePathname: () => mockPath,
    useSearchParams: () => {},
    useRouter: jest.fn().mockImplementation(() => ({
      replace: mockReplace,
    })),
  };
});

const mockSelectedAreasData = [
  mockAreaDataForNHSRegion['E40000007'],
  mockAreaDataForNHSRegion['E40000012'],
];

describe('Area Filter', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('snapshot test', () => {
    const container = render(<AreaFilter availableAreaTypes={[]} />);

    expect(container.asFragment()).toMatchSnapshot();
  });

  it('should render the Filters heading', () => {
    render(<AreaFilter />);

    expect(screen.getByRole('heading')).toHaveTextContent('Filters');
  });

  it('should render the selected areas when there are areas selected', () => {
    render(<AreaFilter selectedAreasData={mockSelectedAreasData} />);

    expect(screen.getByText(/Selected areas \(2\)/i)).toBeInTheDocument();
    expect(screen.getAllByTestId('pill-container')).toHaveLength(2);
  });

  it('should remove the area selected from the url when the remove icon is clicked for the area selected', async () => {
    const expectedPath = [
      `${mockPath}`,
      `?${SearchParams.AreasSelected}=E40000012`,
      `&${SearchParams.AreaTypeSelected}=NHS+region`,
    ].join('');

    const user = userEvent.setup();
    render(
      <AreaFilter
        selectedAreasData={mockSelectedAreasData}
        searchState={{
          [SearchParams.AreasSelected]: ['E40000012', 'E40000007'],
          [SearchParams.AreaTypeSelected]: 'NHS region',
        }}
      />
    );

    const firstSelectedAreaPill = screen.getAllByTestId('pill-container')[0];
    await user.click(
      within(firstSelectedAreaPill).getByTestId('remove-icon-div')
    );

    expect(mockReplace).toHaveBeenCalledWith(expectedPath, {
      scroll: false,
    });
  });

  describe('Area type', () => {
    it('should disable the select area type drop down when there are areas selected', () => {
      render(<AreaFilter selectedAreasData={mockSelectedAreasData} />);

      expect(
        screen.getByRole('combobox', { name: /Select an area type/i })
      ).toBeDisabled();
    });

    it('should not disable the select area type drop down when there are no areas selected', () => {
      render(<AreaFilter selectedAreasData={[]} />);

      expect(
        screen.getByRole('combobox', { name: /Select an area type/i })
      ).not.toBeDisabled();
    });

    it('should render the select area type drop down and indicate there are no areas selected', () => {
      render(<AreaFilter availableAreaTypes={mockAreaTypes} />);

      expect(screen.getByText(/Selected areas \(0\)/i)).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /Select an area type/i })
      ).toBeInTheDocument();
    });

    it('should render all the available areaTypes sorted by level', () => {
      render(<AreaFilter availableAreaTypes={mockAreaTypes} />);

      const areaTypeDropDown = screen.getByRole('combobox', {
        name: /Select an area type/i,
      });

      const allOptions = within(areaTypeDropDown).getAllByRole('option');

      expect(
        screen.getByRole('combobox', { name: /Select an area type/i })
      ).toHaveLength(mockAreaTypes.length);

      allOptions.forEach((option, i) => {
        expect(option.textContent).toEqual(mockAreaTypes[i].name);
      });
    });

    it('should have have the areaTypeSelected as the pre-selected value', () => {
      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          searchState={{
            [SearchParams.AreaTypeSelected]: 'NHS region',
          }}
        />
      );

      expect(
        screen.getByRole('combobox', { name: /Select an area type/i })
      ).toHaveValue('NHS region');
    });

    it('should add the selected areaType to the url', async () => {
      const expectedPath = [
        `${mockPath}`,
        `?${SearchParams.AreasSelected}=E40000012&${SearchParams.AreasSelected}=E40000007`,
        `&${SearchParams.AreaTypeSelected}=NHS+region`,
      ].join('');

      const user = userEvent.setup();
      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          searchState={{
            [SearchParams.AreasSelected]: ['E40000012', 'E40000007'],
          }}
        />
      );

      await user.selectOptions(
        screen.getByRole('combobox', { name: /Select an area type/i }),
        'NHS region'
      );

      expect(mockReplace).toHaveBeenCalledWith(expectedPath, {
        scroll: false,
      });
    });
  });

  describe('Group type', () => {
    it('should disable the select group type drop down when there are areas selected', () => {
      render(<AreaFilter selectedAreasData={mockSelectedAreasData} />);

      expect(
        screen.getByRole('combobox', { name: /1. Select a group type/i })
      ).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /1. Select a group type/i })
      ).toBeDisabled();
    });

    it('should not disable the select group type drop down when there are no areas selected', () => {
      render(<AreaFilter selectedAreasData={[]} />);

      expect(
        screen.getByRole('combobox', { name: /1. Select a group type/i })
      ).toBeInTheDocument();
      expect(
        screen.getByRole('combobox', { name: /1. Select a group type/i })
      ).not.toBeDisabled();
    });

    it('should render all applicable group types based upon the area type selected', () => {
      const expectedAreaTypeOptions = ['Country'];

      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          searchState={{
            [SearchParams.AreaTypeSelected]: 'NHS region',
          }}
        />
      );

      const groupTypeDropDown = screen.getByRole('combobox', {
        name: /1. Select a group type/i,
      });

      const allOptions = within(groupTypeDropDown).getAllByRole('option');

      expect(
        screen.getByRole('combobox', { name: /1. Select a group type/i })
      ).toHaveLength(1);

      allOptions.forEach((option, i) => {
        expect(option.textContent).toEqual(expectedAreaTypeOptions[i]);
      });
    });

    it('should have have the groupTypeSelected as the pre-selected value', () => {
      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          searchState={{
            [SearchParams.AreaTypeSelected]: 'NHS region',
            [SearchParams.GroupTypeSelected]: 'Country',
          }}
        />
      );

      expect(
        screen.getByRole('combobox', { name: /Select a group type/i })
      ).toHaveValue('Country');
    });

    it('should add the selected groupType to the url', async () => {
      const expectedPath = [
        `${mockPath}`,
        `?${SearchParams.AreasSelected}=E40000012&${SearchParams.AreasSelected}=E40000007`,
        `&${SearchParams.AreaTypeSelected}=NHS+region`,
        `&${SearchParams.GroupTypeSelected}=Country`,
      ].join('');

      const user = userEvent.setup();

      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          searchState={{
            [SearchParams.AreasSelected]: ['E40000012', 'E40000007'],
            [SearchParams.AreaTypeSelected]: 'NHS region',
          }}
        />
      );

      await user.selectOptions(
        screen.getByRole('combobox', { name: /Select a group type/i }),
        'Country'
      );

      expect(mockReplace).toHaveBeenCalledWith(expectedPath, {
        scroll: false,
      });
    });
  });

  describe('Areas', () => {
    it('should show all the applicable areas as checkboxes', () => {
      const availableAreas = mockAvailableAreas['NHS region'];

      render(
        <AreaFilter
          availableAreaTypes={mockAreaTypes}
          availableAreas={availableAreas}
          searchState={{
            [SearchParams.AreaTypeSelected]: 'NHS region',
          }}
        />
      );

      expect(screen.getAllByRole('checkbox')).toHaveLength(7);

      availableAreas.forEach((area) => {
        expect(
          screen.getByRole('checkbox', { name: area.name })
        ).toBeInTheDocument();
      });
    });
  });
});
