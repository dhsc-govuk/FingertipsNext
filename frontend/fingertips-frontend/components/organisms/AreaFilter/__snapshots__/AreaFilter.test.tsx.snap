// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area Filter snapshot test 1`] = `
<DocumentFragment>
  .c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  display: block;
  clear: none;
  padding-bottom: 2px;
  margin-bottom: 0;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  margin-bottom: 0;
}

.c13:after {
  content: '';
  display: table;
  clear: both;
}

.c7 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  margin-bottom: 20px;
  display: block;
}

.c10 {
  display: inline-block;
  position: relative;
  margin-bottom: 5px;
  padding-left: 25px;
  color: #1d70b8;
  cursor: pointer;
}

.c10:hover {
  color: #2b8cc4;
}

.c10:focus {
  outline: 4px solid #ffdd00;
  outline-offset: -1px;
  color: #0b0c0c;
  background: #ffdd00;
}

.c10::-webkit-details-marker {
  display: none;
}

.c10:before {
  content: "";
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  margin: auto;
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,100% 50%,0% 100%);
  clip-path: polygon(0% 0%,100% 50%,0% 100%);
  border-width: 7px 0 7px 12.124px;
  border-left-color: inherit;
}

[open] > .c9:before {
  display: block;
  width: 0;
  height: 0;
  border-style: solid;
  border-color: transparent;
  -webkit-clip-path: polygon(0% 0%,50% 100%,100% 0%);
  clip-path: polygon(0% 0%,50% 100%,100% 0%);
  border-width: 12.124px 7px 0 7px;
  border-top-color: inherit;
}

.c11 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c12 {
  padding: 15px;
  padding-left: 20px;
  border-left: 5px solid #bfc1c3;
}

.c12 p {
  margin-top: 0;
  margin-bottom: 20px;
}

.c12 > :last-child,
.c12 p:last-child {
  margin-bottom: 0;
}

.c1 {
  color: #0b0c0c;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 18px;
  line-height: 1.1111111111111112;
  display: block;
  margin-top: 0;
  margin-bottom: 15px;
}

.c17 {
  margin-bottom: 20px;
}

.c16 .c16:last-of-type {
  margin-bottom: 0;
}

.c2 {
  margin: 0;
  border: 0;
  border-bottom: 1px solid #bfc1c3;
}

.c15 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  box-sizing: border-box;
  width: 100%;
  height: 33px;
  padding: 5px 4px 4px;
  border: 2px solid #0b0c0c;
}

.c15:focus {
  outline: 3px solid #ffdd00;
  outline-offset: 0;
}

.c8 div {
  padding: 1em 0em;
}

.c8 summary {
  color: #000000;
}

.c8 div {
  border-left: none;
}

.c0 {
  background-color: #D1D2D3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: -1.3em;
  padding: 0.5em 1em;
}

.c4 {
  padding-bottom: 1.5em;
}

.c3 {
  background-color: #E1E2E3;
  min-height: 100%;
  padding: 1.5em 1em;
}

.c6 {
  font-weight: bold;
}

.c14 {
  margin-bottom: 1em;
}

.c14 span {
  font-weight: bold;
}

.c14 select {
  width: 100%;
}

@media print {
  .c5 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c5 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    margin-bottom: 0;
  }
}

@media only screen and (min-width:641px) {
  .c13 {
    margin-bottom: 0;
  }
}

@media print {
  .c7 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c7 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c7 {
    margin-bottom: 30px;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media print {
  .c1 {
    font-size: 18px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 24px;
    line-height: 1.25;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 20px;
  }
}

@media only screen and (min-width:641px) {
  .c17 {
    margin-bottom: 30px;
  }
}

@media print {
  .c15 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c15 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c15 {
    width: 50%;
    height: 38px;
  }
}

<div
    class=""
    data-testid="area-filter-container"
  >
    <div
      class="c0"
    >
      <h3
        class="c1"
      >
        Filters
      </h3>
    </div>
    <hr
      class="c2"
    />
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <span
          class="c5 c6"
        >
          Selected areas (0)
        </span>
      </div>
      <details
        class="c7 c8"
        open=""
      >
        <summary
          class="c9 c10"
        >
          <span
            class="c11"
          >
            Add or change areas
          </span>
        </summary>
        <div
          class="c12"
        >
          <label
            class="c13 c14"
            data-testid="area-type-selector-container"
          >
            <span
              class="c5"
            >
              Select an area type
            </span>
            <select
              class="c15"
            />
          </label>
          <label
            class="c13 c14"
            data-testid="group-type-selector-container"
          >
            <span
              class="c5"
            >
              Select a group type
            </span>
            <select
              class="c15"
            />
          </label>
          <label
            class="c13 c14"
          >
            <span
              class="c5"
            >
              Select a group
            </span>
            <select
              class="c15"
            />
          </label>
          <div
            class="c16 c17"
          >
            <span
              class="c5 c6"
            >
              Select one or more areas
            </span>
          </div>
        </div>
      </details>
    </div>
  </div>
</DocumentFragment>
`;
